### PR TITLE
Update to recent platform chart version

### PR DESCRIPTION
**Pull request by Zelldon**
# Update Camunda platform dependency

Updating to the recent version involved several changes, as some behaviors and properties changed in the parent chart. See https://docs.camunda.io/docs/self-managed/setup/upgrade/#deprecation-notes

## Prework:

* Introduced a variable in make file, so I can build the project locally. I have multiple go versions / binaries installed. 
* Add missing golden file tests for zeebe deployments

## Migration work:

* Disable identity keycloak it was separated from identity subchart (there exist no longer an sub chart)
* Change zeebe-gateway to zeebeGateway
* Set tag on all components, see related [thread](https://camunda.slack.com/archives/C03UR0V2R2M/p1722937641937199)
  * Fix in parent to use right version in label - https://github.com/camunda/camunda-platform-helm/pull/2183
* Rename the benchmark zeebe-config - application.yaml file and mount to benchmark-config.yaml to not clash with the new default application.yaml
  * **Otherwise**: Installation will be broken. We can't define/mount an `application.yaml` as this is now done using the platform chart. We can't override the default configuration, as this replaces all defaults. We can't copy the template, as templates are not supported currently. We can't simply copy values from an installed chart, as they depend on the configs of the chart and are dynamic evaluated like `clusterName: {{ tpl .Values.global.zeebeClusterName . }}`. See [thread](https://camunda.slack.com/archives/C03UR0V2R2M/p1722947834149849) 
  * In order to load the benchmark-config.yaml we configure spring for an additional config location, see https://docs.spring.io/spring-boot/reference/features/external-config.html#features.external-config.files. Thanks to @npepinpe for the hint.
* Generate all golden files

# Related

Closes https://github.com/zeebe-io/benchmark-helm/pull/158
Closes https://github.com/zeebe-io/benchmark-helm/pull/145

### CODE REVIEW

**Comment by Zelldon** [comment]
With [the](https://github.com/zeebe-io/benchmark-helm/pull/171/commits/35daa48c0a13f7bcc6ee8982b861a8359b02d107) and especially setting `SPRING_CONFIG_ADDITIONALLOCATION` we can now install the chart again. We can also see the config has been applied via:

```
 k port-forward svc/zeebe-benchmark-test-zeebe 9600

```

And accessing http://localhost:9600/actuator/configprops


In the gateway the threads are set

In the brokers the properties are set as well

**Comment by Zelldon** [comment]
Benchmark is successfully running here https://grafana.dev.zeebe.io/d/zeebe-dashboard/zeebe?orgId=1&refresh=10s&var-DS_PROMETHEUS=prometheus&var-cluster=All&var-namespace=ck-update-helm&var-pod=All&var-partition=All&from=now-3h&to=now

**Comment by Zelldon** [comment]
As the aftermath we have to adjust the benchmark helm install settings/configs in the Camunda repo.

**Review by npepinpe — APPROVED** [review]
:+1: Thanks for doing that

:question: Should we enable Identity eventually? We can't benchmark multi-tenancy without it, for example. And we already know there are some performance issues surrounding it (e.g. blocking calls in gateway). Plus, they will start using Zeebe as a DB, so we will likely want that. Not in this scope of course, just saying because you mentioned disabling it :smile: 

:wrench: Did you start a line with upstream about extensible configuration? An issue, Slack thread, or something?

File: charts/zeebe-benchmark/values.yaml [hunk 1]
```diff
@@ -223,6 +224,8 @@ camunda-platform:
         value: "0"
       - name: PYROSCOPE_LABELS
         value: "namespace=$(K8S_NAMESPACE), pod=$(K8S_NAME)"
+      - name: SPRING_CONFIG_ADDITIONALLOCATION
```
Comment (npepinpe) [inline charts/zeebe-benchmark/values.yaml:227]: :clown_face: I keep reading this as Addition Allocation :smile:

Comment (lenaschoenburg) [reply charts/zeebe-benchmark/values.yaml:227]: Oh my, I was wondering what the hell spring wants to allocate until I read your comment :facepalm:

**Review by lenaschoenburg — APPROVED** [review]

**Comment by Zelldon** [comment]
> ❓ Should we enable Identity eventually? We can't benchmark multi-tenancy without it, for example. And we already know there are some performance issues surrounding it (e.g. blocking calls in gateway). Plus, they will start using Zeebe as a DB, so we will likely want that. Not in this scope of course, just saying because you mentioned disabling it 😄

Thanks for the question. I don't think we need to do this in the defaults for now. This can be done via `--set camunda-platform.identity.enabled=true` for example. I actually planned to do once a chaos day with identity, but had never time for it, unfortunately. 

> 🔧 Did you start a line with upstream about extensible configuration? An issue, Slack thread, or something?

See slack thread https://camunda.slack.com/archives/C03UR0V2R2M/p1722947834149849
See github issue https://github.com/camunda/camunda-platform-helm/issues/2197

File: charts/zeebe-benchmark/values.yaml (hunk 1 above)
Comment (Zelldon) [reply charts/zeebe-benchmark/values.yaml:229]: Thanks for the input you both :) I added an additional comment to clarify this a bit.

**Pull request merged**